### ISSUE TITLE
Vector not behaving when used within a query in cljs 

### ISSUE THREAD

**Issue by bahulneel** [tonsky/datascript#262]
Hi,

I get some odd behaviour when using the vector function within a query.

Take the following query:
```clojure
(datascript.core/q
  '{:find [?product-type ?v]
    :where
          ([?id-10 :product-type ?product-type] 
           [(vector ?product-type) ?v])}

  (-> (datascript.core/empty-db)
      (datascript.core/with [{:product-type "A"}
                             {:product-type "B"}])
      :db-after))
=> #{["A" ["B"]] ["B" ["B"]]}
```

I initially thought this was due to function calls, however, when I provide a different implementation of vector then the problem goes away:

```clojure
(datascript.core/q
  '{:find [?product-type ?v]
    :in [$ ?vector]
    :where
          ([?id-10 :product-type ?product-type] 
           [(?vector ?product-type) ?v])}

  (-> (datascript.core/empty-db)
      (datascript.core/with [{:product-type "A"}
                             {:product-type "B"}])
      :db-after)
  (fn [& args] (vec args)))
=> #{["A" ["A"]] ["B" ["B"]]}
```

NOTE: This does not happen with the clj version, only with cljs (on node).

**Comment by cgrand** [comment]
Awesome, thanks!

Looking at your patch, shouldn't it be fixed in CLJS? `apply vector` shouldn't behave like `LazilyPersistentVector/createOwning`.

**Comment by tonsky** [comment]
I don’t know — given that they spent so much effort making sure vector does hold onto passed mutable array, and that Clojure does almost the same in `vec`, I doubt they’ll be willing to change that. 

Also I’m not sure if that’s the issue at all — it all only starts when you pass mutable arrays to Clojure/Script and then change them underneath. I suspect it might be an unspoken understanding that if you pass a mutable array and start changing it later that might lead to undefined behaviour.

TL;DR I can’t even formulate why it is the problem :)

**Comment by cgrand** [comment]
Opened a ticket https://clojure.atlassian.net/browse/CLJS-3095

**Comment by mfikes** [comment]
A fix for the underlying ClojureScript issue referenced above should be in the ClojureScript 1.10.597 release.
